File: lib/cfr/builtins/libs/dart/ui/platformDispatcher.dart
import 'dart:core';
import 'dart:typed_data';
import 'dart:ui';

import 'package:hydro_sdk/cfr/runtimeSupport.dart';

class VMManagedPlatformDispatcher extends VMManagedBox<PlatformDispatcher> {
  VMManagedPlatformDispatcher(
      {required this.table, required this.vmObject, required this.hydroState})
      : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) {
    table['getConfiguration'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<PlatformConfiguration>(
            object: vmObject.configuration,
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['getOnPlatformConfigurationChanged'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      final returnValue = vmObject.onPlatformConfigurationChanged;
      if (returnValue != null) {
        return [
          returnValue,
        ];
      }
      return [];
    });
    table['setOnPlatformConfigurationChanged'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      Closure? unpackedcallback = luaCallerArguments[1];
      vmObject.onPlatformConfigurationChanged = (unpackedcallback != null
          ? () => unpackedcallback.dispatch(
                [
                  luaCallerArguments[0],
                ],
                parentState: hydroState,
              )
          : null);
      return [];
    });
    table['getViews'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Iterable>(
            object: vmObject.views,
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['getOnMetricsChanged'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      final returnValue = vmObject.onMetricsChanged;
      if (returnValue != null) {
        return [
          returnValue,
        ];
      }
      return [];
    });
    table['setOnMetricsChanged'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      Closure? unpackedcallback = luaCallerArguments[1];
      vmObject.onMetricsChanged = (unpackedcallback != null
          ? () => unpackedcallback.dispatch(
                [
                  luaCallerArguments[0],
                ],
                parentState: hydroState,
              )
          : null);
      return [];
    });
    table['getOnBeginFrame'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      final returnValue = vmObject.onBeginFrame;
      if (returnValue != null) {
        return [
          returnValue,
        ];
      }
      return [];
    });
    table['setOnBeginFrame'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      Closure? unpackedcallback = luaCallerArguments[1];
      vmObject.onBeginFrame = (unpackedcallback != null
          ? (duration) => unpackedcallback.dispatch(
                [luaCallerArguments[0], duration],
                parentState: hydroState,
              )
          : null);
      return [];
    });
    table['getOnDrawFrame'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      final returnValue = vmObject.onDrawFrame;
      if (returnValue != null) {
        return [
          returnValue,
        ];
      }
      return [];
    });
    table['setOnDrawFrame'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      Closure? unpackedcallback = luaCallerArguments[1];
      vmObject.onDrawFrame = (unpackedcallback != null
          ? () => unpackedcallback.dispatch(
                [
                  luaCallerArguments[0],
                ],
                parentState: hydroState,
              )
          : null);
      return [];
    });
    table['getOnPointerDataPacket'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      final returnValue = vmObject.onPointerDataPacket;
      if (returnValue != null) {
        return [
          returnValue,
        ];
      }
      return [];
    });
    table['setOnPointerDataPacket'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      Closure? unpackedcallback = luaCallerArguments[1];
      vmObject.onPointerDataPacket = (unpackedcallback != null
          ? (packet) => unpackedcallback.dispatch(
                [luaCallerArguments[0], packet],
                parentState: hydroState,
              )
          : null);
      return [];
    });
    table['getOnReportTimings'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      final returnValue = vmObject.onReportTimings;
      if (returnValue != null) {
        return [
          returnValue,
        ];
      }
      return [];
    });
    table['setOnReportTimings'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      Closure? unpackedcallback = luaCallerArguments[1];
      vmObject.onReportTimings = (unpackedcallback != null
          ? (timings) => unpackedcallback.dispatch(
                [luaCallerArguments[0], timings],
                parentState: hydroState,
              )
          : null);
      return [];
    });
    table['sendPlatformMessage'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      Closure? unpackedcallback = luaCallerArguments[3];
      vmObject.sendPlatformMessage(
          luaCallerArguments[1],
          maybeUnBoxAndBuildArgument<ByteData?, dynamic>(luaCallerArguments[2],
              parentState: hydroState),
          unpackedcallback != null
              ? (data) => unpackedcallback.dispatch(
                    [luaCallerArguments[0], data],
                    parentState: hydroState,
                  )
              : null);
      return [];
    });
    table['getOnPlatformMessage'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      final returnValue = vmObject.onPlatformMessage;
      if (returnValue != null) {
        return [
          returnValue,
        ];
      }
      return [];
    });
    table['setOnPlatformMessage'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      Closure? unpackedcallback = luaCallerArguments[1];
      vmObject.onPlatformMessage = (unpackedcallback != null
          ? (name, data, callback) => unpackedcallback.dispatch(
                [luaCallerArguments[0], name, data, callback],
                parentState: hydroState,
              )
          : null);
      return [];
    });
    table['setIsolateDebugName'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      vmObject.setIsolateDebugName(luaCallerArguments[1]);
      return [];
    });
    table['getPersistentIsolateData'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      final returnValue = vmObject.getPersistentIsolateData();
      if (returnValue != null) {
        return [
          maybeBoxObject<ByteData?>(
              object: returnValue, hydroState: hydroState, table: HydroTable()),
        ];
      }
      return [];
    });
    table['scheduleFrame'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      vmObject.scheduleFrame();
      return [];
    });
    table['getAccessibilityFeatures'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<AccessibilityFeatures>(
            object: vmObject.accessibilityFeatures,
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['getOnAccessibilityFeaturesChanged'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      final returnValue = vmObject.onAccessibilityFeaturesChanged;
      if (returnValue != null) {
        return [
          returnValue,
        ];
      }
      return [];
    });
    table['setOnAccessibilityFeaturesChanged'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      Closure? unpackedcallback = luaCallerArguments[1];
      vmObject.onAccessibilityFeaturesChanged = (unpackedcallback != null
          ? () => unpackedcallback.dispatch(
                [
                  luaCallerArguments[0],
                ],
                parentState: hydroState,
              )
          : null);
      return [];
    });
    table['updateSemantics'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      vmObject.updateSemantics(
          maybeUnBoxAndBuildArgument<SemanticsUpdate, dynamic>(
              luaCallerArguments[1],
              parentState: hydroState));
      return [];
    });
    table['getLocale'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Locale>(
            object: vmObject.locale,
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['getLocales'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<List<dynamic>>(
            object: vmObject.locales
                .map((x) => maybeBoxObject<Locale>(
                    object: x, hydroState: hydroState, table: HydroTable()))
                .toList(),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['computePlatformResolvedLocale'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      final returnValue = vmObject.computePlatformResolvedLocale(
          maybeUnBoxAndBuildArgument<List<Locale>, Locale>(
              luaCallerArguments[1],
              parentState: hydroState));
      if (returnValue != null) {
        return [
          maybeBoxObject<Locale?>(
              object: returnValue, hydroState: hydroState, table: HydroTable()),
        ];
      }
      return [];
    });
    table['getOnLocaleChanged'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      final returnValue = vmObject.onLocaleChanged;
      if (returnValue != null) {
        return [
          returnValue,
        ];
      }
      return [];
    });
    table['setOnLocaleChanged'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      Closure? unpackedcallback = luaCallerArguments[1];
      vmObject.onLocaleChanged = (unpackedcallback != null
          ? () => unpackedcallback.dispatch(
                [
                  luaCallerArguments[0],
                ],
                parentState: hydroState,
              )
          : null);
      return [];
    });
    table['getInitialLifecycleState'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.initialLifecycleState,
      ];
    });
    table['getAlwaysUse24HourFormat'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.alwaysUse24HourFormat,
      ];
    });
    table['getTextScaleFactor'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.textScaleFactor,
      ];
    });
    table['getOnTextScaleFactorChanged'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      final returnValue = vmObject.onTextScaleFactorChanged;
      if (returnValue != null) {
        return [
          returnValue,
        ];
      }
      return [];
    });
    table['setOnTextScaleFactorChanged'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      Closure? unpackedcallback = luaCallerArguments[1];
      vmObject.onTextScaleFactorChanged = (unpackedcallback != null
          ? () => unpackedcallback.dispatch(
                [
                  luaCallerArguments[0],
                ],
                parentState: hydroState,
              )
          : null);
      return [];
    });
    table['getPlatformBrightness'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        Brightness.values.indexWhere((x) {
          return x == vmObject.platformBrightness;
        }),
      ];
    });
    table['getOnPlatformBrightnessChanged'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      final returnValue = vmObject.onPlatformBrightnessChanged;
      if (returnValue != null) {
        return [
          returnValue,
        ];
      }
      return [];
    });
    table['setOnPlatformBrightnessChanged'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      Closure? unpackedcallback = luaCallerArguments[1];
      vmObject.onPlatformBrightnessChanged = (unpackedcallback != null
          ? () => unpackedcallback.dispatch(
                [
                  luaCallerArguments[0],
                ],
                parentState: hydroState,
              )
          : null);
      return [];
    });
    table['getSemanticsEnabled'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.semanticsEnabled,
      ];
    });
    table['getOnSemanticsEnabledChanged'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      final returnValue = vmObject.onSemanticsEnabledChanged;
      if (returnValue != null) {
        return [
          returnValue,
        ];
      }
      return [];
    });
    table['setOnSemanticsEnabledChanged'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      Closure? unpackedcallback = luaCallerArguments[1];
      vmObject.onSemanticsEnabledChanged = (unpackedcallback != null
          ? () => unpackedcallback.dispatch(
                [
                  luaCallerArguments[0],
                ],
                parentState: hydroState,
              )
          : null);
      return [];
    });
    table['getOnSemanticsAction'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      final returnValue = vmObject.onSemanticsAction;
      if (returnValue != null) {
        return [
          returnValue,
        ];
      }
      return [];
    });
    table['setOnSemanticsAction'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      Closure? unpackedcallback = luaCallerArguments[1];
      vmObject.onSemanticsAction = (unpackedcallback != null
          ? (id, action, args) => unpackedcallback.dispatch(
                [luaCallerArguments[0], id, action, args],
                parentState: hydroState,
              )
          : null);
      return [];
    });
    table['getDefaultRouteName'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.defaultRouteName,
      ];
    });
  }

  final HydroTable table;

  final HydroState hydroState;

  final PlatformDispatcher vmObject;
}

void loadPlatformDispatcher(
    {required HydroState hydroState, required HydroTable table}) {
  table['platformDispatcherInstance'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<PlatformDispatcher>(
          object: PlatformDispatcher.instance,
          hydroState: hydroState,
          table: HydroTable()),
    ];
  });
  registerBoxer<PlatformDispatcher>(boxer: (
      {required PlatformDispatcher vmObject,
      required HydroState hydroState,
      required HydroTable table}) {
    return VMManagedPlatformDispatcher(
        vmObject: vmObject, hydroState: hydroState, table: table);
  });
}

File: runtime/dart/ui/platformDispatcher.ts
import { IDuration } from "../core/duration";
import { IIterable } from "../core/iterable";
import { IList } from "../core/list";
import { IByteData } from "../typed_data/byteData";
import { IAccessibilityFeatures } from "./accessibilityFeatures";
import { Brightness } from "./brightness";
import { IFlutterView } from "./flutterView";
import { IFrameTiming } from "./frameTiming";
import { ILocale } from "./locale";
import { IPlatformConfiguration } from "./platformConfiguration";
import { IPointerDataPacket } from "./pointerDataPacket";
import { ISemanticsAction } from "./semanticsAction";
import { ISemanticsUpdate } from "./semanticsUpdate";
declare const dart: {
    ui: {
        platformDispatcherInstance: () => IPlatformDispatcher;
    };
};
export interface IPlatformDispatcher {
    getConfiguration: () => IPlatformConfiguration;
    getOnPlatformConfigurationChanged: () => () => void | undefined;
    setOnPlatformConfigurationChanged: (callback?: () => void) => void;
    getViews: () => IIterable<IFlutterView>;
    getOnMetricsChanged: () => () => void | undefined;
    setOnMetricsChanged: (callback?: () => void) => void;
    getOnBeginFrame: () => (duration: IDuration) => void | undefined;
    setOnBeginFrame: (callback?: (duration: IDuration) => void) => void;
    getOnDrawFrame: () => () => void | undefined;
    setOnDrawFrame: (callback?: () => void) => void;
    getOnPointerDataPacket: () => (
        packet: IPointerDataPacket
    ) => void | undefined;
    setOnPointerDataPacket: (
        callback?: (packet: IPointerDataPacket) => void
    ) => void;
    getOnReportTimings: () => (
        timings: IList<IFrameTiming>
    ) => void | undefined;
    setOnReportTimings: (
        callback?: (timings: IList<IFrameTiming>) => void
    ) => void;
    sendPlatformMessage: (
        name: string,
        data?: IByteData | undefined,
        callback?: (data?: IByteData | undefined) => void
    ) => void;
    getOnPlatformMessage: () => (
        name: string,
        data?: IByteData | undefined,
        callback?: (data?: IByteData | undefined) => void
    ) => void | undefined;
    setOnPlatformMessage: (
        callback?: (
            name: string,
            data?: IByteData | undefined,
            callback?: (data?: IByteData | undefined) => void
        ) => void
    ) => void;
    setIsolateDebugName: (name: string) => void;
    getPersistentIsolateData: () => IByteData | undefined;
    scheduleFrame: () => void;
    getAccessibilityFeatures: () => IAccessibilityFeatures;
    getOnAccessibilityFeaturesChanged: () => () => void | undefined;
    setOnAccessibilityFeaturesChanged: (callback?: () => void) => void;
    updateSemantics: (update: ISemanticsUpdate) => void;
    getLocale: () => ILocale;
    getLocales: () => IList<ILocale>;
    computePlatformResolvedLocale: (
        supportedLocales: IList<ILocale>
    ) => ILocale | undefined;
    getOnLocaleChanged: () => () => void | undefined;
    setOnLocaleChanged: (callback?: () => void) => void;
    getInitialLifecycleState: () => string;
    getAlwaysUse24HourFormat: () => boolean;
    getTextScaleFactor: () => number;
    getOnTextScaleFactorChanged: () => () => void | undefined;
    setOnTextScaleFactorChanged: (callback?: () => void) => void;
    getPlatformBrightness: () => Brightness;
    getOnPlatformBrightnessChanged: () => () => void | undefined;
    setOnPlatformBrightnessChanged: (callback?: () => void) => void;
    getSemanticsEnabled: () => boolean;
    getOnSemanticsEnabledChanged: () => () => void | undefined;
    setOnSemanticsEnabledChanged: (callback?: () => void) => void;
    getOnSemanticsAction: () => (
        id: number,
        action: ISemanticsAction,
        args?: IByteData | undefined
    ) => void | undefined;
    setOnSemanticsAction: (
        callback?: (
            id: number,
            action: ISemanticsAction,
            args?: IByteData | undefined
        ) => void
    ) => void;
    getDefaultRouteName: () => string;
}
export class PlatformDispatcher {
    public static _kPointerDataFieldCount = 29;
    public static instance(): IPlatformDispatcher {
        return dart.ui.platformDispatcherInstance();
    }
    private readonly _dart_getConfiguration: () => IPlatformConfiguration =
        undefined as any;
    private readonly _dart_getOnPlatformConfigurationChanged: () => () =>
        | void
        | undefined = undefined as any;
    private readonly _dart_setOnPlatformConfigurationChanged: (
        callback?: () => void
    ) => void = undefined as any;
    private readonly _dart_getViews: () => IIterable<IFlutterView> =
        undefined as any;
    private readonly _dart_getOnMetricsChanged: () => () => void | undefined =
        undefined as any;
    private readonly _dart_setOnMetricsChanged: (
        callback?: () => void
    ) => void = undefined as any;
    private readonly _dart_getOnBeginFrame: () => (
        duration: IDuration
    ) => void | undefined = undefined as any;
    private readonly _dart_setOnBeginFrame: (
        callback?: (duration: IDuration) => void
    ) => void = undefined as any;
    private readonly _dart_getOnDrawFrame: () => () => void | undefined =
        undefined as any;
    private readonly _dart_setOnDrawFrame: (callback?: () => void) => void =
        undefined as any;
    private readonly _dart_getOnPointerDataPacket: () => (
        packet: IPointerDataPacket
    ) => void | undefined = undefined as any;
    private readonly _dart_setOnPointerDataPacket: (
        callback?: (packet: IPointerDataPacket) => void
    ) => void = undefined as any;
    private readonly _dart_getOnReportTimings: () => (
        timings: IList<IFrameTiming>
    ) => void | undefined = undefined as any;
    private readonly _dart_setOnReportTimings: (
        callback?: (timings: IList<IFrameTiming>) => void
    ) => void = undefined as any;
    private readonly _dart_sendPlatformMessage: (
        name: string,
        data?: IByteData | undefined,
        callback?: (data?: IByteData | undefined) => void
    ) => void = undefined as any;
    private readonly _dart_getOnPlatformMessage: () => (
        name: string,
        data?: IByteData | undefined,
        callback?: (data?: IByteData | undefined) => void
    ) => void | undefined = undefined as any;
    private readonly _dart_setOnPlatformMessage: (
        callback?: (
            name: string,
            data?: IByteData | undefined,
            callback?: (data?: IByteData | undefined) => void
        ) => void
    ) => void = undefined as any;
    private readonly _dart_setIsolateDebugName: (name: string) => void =
        undefined as any;
    private readonly _dart_getPersistentIsolateData: () =>
        | IByteData
        | undefined = undefined as any;
    private readonly _dart_scheduleFrame: () => void = undefined as any;
    private readonly _dart_getAccessibilityFeatures: () => IAccessibilityFeatures =
        undefined as any;
    private readonly _dart_getOnAccessibilityFeaturesChanged: () => () =>
        | void
        | undefined = undefined as any;
    private readonly _dart_setOnAccessibilityFeaturesChanged: (
        callback?: () => void
    ) => void = undefined as any;
    private readonly _dart_updateSemantics: (update: ISemanticsUpdate) => void =
        undefined as any;
    private readonly _dart_getLocale: () => ILocale = undefined as any;
    private readonly _dart_getLocales: () => IList<ILocale> = undefined as any;
    private readonly _dart_computePlatformResolvedLocale: (
        supportedLocales: IList<ILocale>
    ) => ILocale | undefined = undefined as any;
    private readonly _dart_getOnLocaleChanged: () => () => void | undefined =
        undefined as any;
    private readonly _dart_setOnLocaleChanged: (callback?: () => void) => void =
        undefined as any;
    private readonly _dart_getInitialLifecycleState: () => string =
        undefined as any;
    private readonly _dart_getAlwaysUse24HourFormat: () => boolean =
        undefined as any;
    private readonly _dart_getTextScaleFactor: () => number = undefined as any;
    private readonly _dart_getOnTextScaleFactorChanged: () => () =>
        | void
        | undefined = undefined as any;
    private readonly _dart_setOnTextScaleFactorChanged: (
        callback?: () => void
    ) => void = undefined as any;
    private readonly _dart_getPlatformBrightness: () => Brightness =
        undefined as any;
    private readonly _dart_getOnPlatformBrightnessChanged: () => () =>
        | void
        | undefined = undefined as any;
    private readonly _dart_setOnPlatformBrightnessChanged: (
        callback?: () => void
    ) => void = undefined as any;
    private readonly _dart_getSemanticsEnabled: () => boolean =
        undefined as any;
    private readonly _dart_getOnSemanticsEnabledChanged: () => () =>
        | void
        | undefined = undefined as any;
    private readonly _dart_setOnSemanticsEnabledChanged: (
        callback?: () => void
    ) => void = undefined as any;
    private readonly _dart_getOnSemanticsAction: () => (
        id: number,
        action: ISemanticsAction,
        args?: IByteData | undefined
    ) => void | undefined = undefined as any;
    private readonly _dart_setOnSemanticsAction: (
        callback?: (
            id: number,
            action: ISemanticsAction,
            args?: IByteData | undefined
        ) => void
    ) => void = undefined as any;
    private readonly _dart_getDefaultRouteName: () => string = undefined as any;
    public getConfiguration(): IPlatformConfiguration {
        return this._dart_getConfiguration();
    }
    public getOnPlatformConfigurationChanged(): () => void | undefined {
        return this._dart_getOnPlatformConfigurationChanged();
    }
    public setOnPlatformConfigurationChanged(callback?: () => void): void {
        return this._dart_setOnPlatformConfigurationChanged(callback);
    }
    public getViews(): IIterable<IFlutterView> {
        return this._dart_getViews();
    }
    public getOnMetricsChanged(): () => void | undefined {
        return this._dart_getOnMetricsChanged();
    }
    public setOnMetricsChanged(callback?: () => void): void {
        return this._dart_setOnMetricsChanged(callback);
    }
    public getOnBeginFrame(): (duration: IDuration) => void | undefined {
        return this._dart_getOnBeginFrame();
    }
    public setOnBeginFrame(callback?: (duration: IDuration) => void): void {
        return this._dart_setOnBeginFrame(callback);
    }
    public getOnDrawFrame(): () => void | undefined {
        return this._dart_getOnDrawFrame();
    }
    public setOnDrawFrame(callback?: () => void): void {
        return this._dart_setOnDrawFrame(callback);
    }
    public getOnPointerDataPacket(): (
        packet: IPointerDataPacket
    ) => void | undefined {
        return this._dart_getOnPointerDataPacket();
    }
    public setOnPointerDataPacket(
        callback?: (packet: IPointerDataPacket) => void
    ): void {
        return this._dart_setOnPointerDataPacket(callback);
    }
    public getOnReportTimings(): (
        timings: IList<IFrameTiming>
    ) => void | undefined {
        return this._dart_getOnReportTimings();
    }
    public setOnReportTimings(
        callback?: (timings: IList<IFrameTiming>) => void
    ): void {
        return this._dart_setOnReportTimings(callback);
    }
    public sendPlatformMessage(
        name: string,
        data?: IByteData | undefined,
        callback?: (data?: IByteData | undefined) => void
    ): void {
        return this._dart_sendPlatformMessage(name, data, callback);
    }
    public getOnPlatformMessage(): (
        name: string,
        data?: IByteData | undefined,
        callback?: (data?: IByteData | undefined) => void
    ) => void | undefined {
        return this._dart_getOnPlatformMessage();
    }
    public setOnPlatformMessage(
        callback?: (
            name: string,
            data?: IByteData | undefined,
            callback?: (data?: IByteData | undefined) => void
        ) => void
    ): void {
        return this._dart_setOnPlatformMessage(callback);
    }
    public setIsolateDebugName(name: string): void {
        return this._dart_setIsolateDebugName(name);
    }
    public getPersistentIsolateData(): IByteData | undefined {
        return this._dart_getPersistentIsolateData();
    }
    public scheduleFrame(): void {
        return this._dart_scheduleFrame();
    }
    public getAccessibilityFeatures(): IAccessibilityFeatures {
        return this._dart_getAccessibilityFeatures();
    }
    public getOnAccessibilityFeaturesChanged(): () => void | undefined {
        return this._dart_getOnAccessibilityFeaturesChanged();
    }
    public setOnAccessibilityFeaturesChanged(callback?: () => void): void {
        return this._dart_setOnAccessibilityFeaturesChanged(callback);
    }
    public updateSemantics(update: ISemanticsUpdate): void {
        return this._dart_updateSemantics(update);
    }
    public getLocale(): ILocale {
        return this._dart_getLocale();
    }
    public getLocales(): IList<ILocale> {
        return this._dart_getLocales();
    }
    public computePlatformResolvedLocale(
        supportedLocales: IList<ILocale>
    ): ILocale | undefined {
        return this._dart_computePlatformResolvedLocale(supportedLocales);
    }
    public getOnLocaleChanged(): () => void | undefined {
        return this._dart_getOnLocaleChanged();
    }
    public setOnLocaleChanged(callback?: () => void): void {
        return this._dart_setOnLocaleChanged(callback);
    }
    public getInitialLifecycleState(): string {
        return this._dart_getInitialLifecycleState();
    }
    public getAlwaysUse24HourFormat(): boolean {
        return this._dart_getAlwaysUse24HourFormat();
    }
    public getTextScaleFactor(): number {
        return this._dart_getTextScaleFactor();
    }
    public getOnTextScaleFactorChanged(): () => void | undefined {
        return this._dart_getOnTextScaleFactorChanged();
    }
    public setOnTextScaleFactorChanged(callback?: () => void): void {
        return this._dart_setOnTextScaleFactorChanged(callback);
    }
    public getPlatformBrightness(): Brightness {
        return this._dart_getPlatformBrightness();
    }
    public getOnPlatformBrightnessChanged(): () => void | undefined {
        return this._dart_getOnPlatformBrightnessChanged();
    }
    public setOnPlatformBrightnessChanged(callback?: () => void): void {
        return this._dart_setOnPlatformBrightnessChanged(callback);
    }
    public getSemanticsEnabled(): boolean {
        return this._dart_getSemanticsEnabled();
    }
    public getOnSemanticsEnabledChanged(): () => void | undefined {
        return this._dart_getOnSemanticsEnabledChanged();
    }
    public setOnSemanticsEnabledChanged(callback?: () => void): void {
        return this._dart_setOnSemanticsEnabledChanged(callback);
    }
    public getOnSemanticsAction(): (
        id: number,
        action: ISemanticsAction,
        args?: IByteData | undefined
    ) => void | undefined {
        return this._dart_getOnSemanticsAction();
    }
    public setOnSemanticsAction(
        callback?: (
            id: number,
            action: ISemanticsAction,
            args?: IByteData | undefined
        ) => void
    ): void {
        return this._dart_setOnSemanticsAction(callback);
    }
    public getDefaultRouteName(): string {
        return this._dart_getDefaultRouteName();
    }
}
